### ISSUE TITLE
Cannot fetch a descriptor with a checksum

### ISSUE THREAD

**Issue by maximilliangeorge** [bitcoinerlab/discovery#5]
Finally got around to trying these libraries out. They look really promising, thanks for putting so much work into this.

Now, whenever I fetch a descriptor with a trailing checksum it doesn't return anything. If omit the checksum the fetch works as expected.

The checksum validation seems to work though (I get appropriate errors for malformed or invalid checksums).

```typescript
const descriptor = 'wsh(or_b(pk(tpubD6NzVbkrYhZ4YBvAzHkE42GAQhWUFDLZFAfUFE9NQ5QfL1XSzD5LUtuwHkLXKDrJmagzHsEGUMkRYwmrDGiNdgLWCCwSjDfWS4nwMzDYRif),s:pk(tpubD6NzVbkrYhZ4XWrepmiT2rk6nM4yr8hHiubA7J3QLTDtkWwvmBfFQ5qsq7nJoERBu9xa2pu8RF1HB1KVgexGw4cGeAkJca55v6MUaXhiFD1)))'

// without checksum

await discovery.fetch({ descriptor })

discovery.whenFetched({ descriptor }) // { fetching: false, timeFetched: 1714320793 }
discovery.getUtxos({ descriptor }) // ['00aa766710dc91b488d391b33f0595984b14e68f0059f6e86280094f458204e9:0']

// with checksum

const descriptorWithChecksum = descriptor + '#fr7n53lr'

await discovery.fetch({ descriptorWithChecksum })

discovery.whenFetched({ descriptorWithChecksum }) // undefined
discovery.getUtxos({ descriptorWithChecksum }) // Error: Cannot derive data from <descriptor>/undefined since it has not been previously fetched

```

**Comment by landabaso** [comment]
Thanks for opening the issue @maximilliangeorge . I'm pushing a fix soon.

**Comment by landabaso** [comment]
I've just published the fix in version 1.0.4 on npm. Please give it a try and let me know if it resolves the issue. Thanks again for reporting this!
